src/config.py

```
@@ -0,0 +1,220 @@
+            full_config['zones']['structures'] = config.structures
+            
+            # Save unified configuration
+            with open(config_path, 'w') as f:
+                json.dump(full_config, f, indent=2)
+            
+            # Reload to refresh internal state
+            self._full_config = full_config
+            _logger.info(f"Configuration saved to {config_path}")
+            
+        except Exception as e:
+            _logger.error(f"Error saving configuration: {e}")
```

Ruff (0.14.4)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)